parsl/executors/high_throughput/executor.py

```diff
@@ -240,32 +245,34 @@ def _queue_management_worker(self):
 
             else:
 
-                if msg is None:
+                if msgs is None:
                     logger.debug("[MTHREAD] Got None, exiting")
                     return
 
                 else:
-                    try:
-                        tid = msg['task_id']
-                    except Exception as e:
-                        raise BadMessage("Message received does not contain 'task_id' field")
+                    for serialized_msg in msgs:
+                        try:
+                            msg = pickle.loads(serialized_msg)
+                            tid = msg['task_id']
+                        except Exception as e:
+                            raise BadMessage("Message received does not contain 'task_id' field")
```

Comment:
That's not the only reason we'd get an Exception? Most likely pickle.loads() would throw something too?

Test membership explicitly or fix the error message, for the sake of confused people later.

(fix not critical for functionality)

Reply:
True, pickling could fail. I'm adding a variant of the BadMessage exception for the unpickling failure. At this stage we are only unpickling {'task_id' : <>, 'result' : <serialized buffer>, 'exception' : <serialized buffer>} messages from the managers, so if this fails, we definitely got a bad message.